charts/datadog/ci/agent-otel-collector-values.yaml

```diff
@@ -0,0 +1,24 @@
+targetSystem: "linux"
```

Comment:
I don't know why it was green. But the new OTEL values.yaml trigger an agent crashloopbackoff

```
52s         Warning   Failed              pod/datadog-3rqb0k8mej-lw4w6                             spec.containers{otel-agent}        kubelet, chart-testing-worker   Error: failed to create containerd task: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec: "otel-agent": executable file not found in $PATH: unknown   53s          2       datadog-3rqb0k8mej-lw4w6.17e1433edb194c8f
```

I suppose it is because in your [test file](https://github.com/DataDog/helm-charts/pull/1429/files#diff-0447168e1eea08cb8e8ac86ec9e814d61f3c4913334cc5f17e031dee53ac411c), the new image tag is not set.

So either you modify the `image-path` [function](https://github.com/DataDog/helm-charts/blob/81599dfe4d0176bd9b1f3e7c3ebe9e1f07a868d6/charts/datadog/templates/_helpers.tpl#L294-L315) to taking into consideration otelCollector.enabled
or your explicitly set the image tag to us the one that container the otel-agent binary....
see: https://github.com/DataDog/helm-charts/actions/runs/9898182121/job/27344411927?pr=1429#step:6:136

Reply:
I have updated to use agent nightly otel build. The container is starting and failing with a different error. 
```
kubectl --request-timeout=30s logs datadog-hby8t2xr27-g585n --namespace datadog-hby8t2xr27 --container otel-agent
2024-07-15 17:28:30 UTC | OTELCOL | INFO | (pkg/util/log/log.go:831 in func1) | 3 Features detected from environment: kubernetes,cri,containerd
2024-07-15 17:28:30 UTC | OTELCOL | INFO | (pkg/util/log/log.go:831 in func1) | Loaded configuration: 
2024-07-15 17:28:30 UTC | OTELCOL | INFO | (pkg/util/log/log.go:831 in func1) | 'site' and 'apm_config.apm_dd_url' are both set in config: setting main endpoint to 'apm_config.apm_dd_url': "https://trace.agent.datadoghq.com/"
2024-07-15 17:28:30 UTC | OTELCOL | INFO | (pkg/util/log/log.go:831 in func1) | Found APM feature flags: [enable_otlp_compute_top_level_by_span_kind]
2024-07-15 17:28:30 UTC | OTELCOL | INFO | (pkg/util/log/log.go:831 in func1) | Could not get hostname via gRPC: context deadline exceeded. Falling back to other methods.
2024-07-15 17:28:30 UTC | OTELCOL | INFO | (pkg/util/log/log.go:831 in func1) | Acquired hostname from OS: "datadog-hby8t2xr27-g585n". Core agent was unreachable at "/opt/datadog-agent/bin/agent/agent": exit status 255.
2024-07-15 17:28:30 UTC | OTELCOL | INFO | (comp/forwarder/defaultforwarder/forwarder.go:46 in NewForwarder) | Retry queue storage on disk is disabled
```

Reply:
I have updated to use agent's nightly build of OTel Agent. can you please review again

Reply:
what will be the final tag format? 
for example `7.56.0-ot` ?

Reply:
It would be `7.56.0-ot-beta`. we have created these public images ```gcr.io/datadoghq/agent:7-ot-beta-jmx
public.ecr.aws/datadog/agent:7-ot-beta-jmx
europe-docker.pkg.dev/datadoghq/eu.gcr.io/agent:7-ot-beta-jmx
asia-docker.pkg.dev/datadoghq/asia.gcr.io/agent:7-ot-beta-jmx```